lib/saml_config.rb

```diff
@@ -2,12 +2,10 @@
 
 module Saml
   class Config
-    def initialize
-      @@settings ||= build_settings.freeze
-    end
+    attr_reader :settings
 
-    def settings
-      @@settings
+    def initialize
+      @settings ||= build_settings.freeze
```

Comment:
@jessieay You can test this by adding the line `puts 'build_settings called'` to the top of the `build_settings` method, and then running `rake`. You will notice that with the class variable version of the code, "build_settings called" only gets output once. With an instance variable, it gets called many times.